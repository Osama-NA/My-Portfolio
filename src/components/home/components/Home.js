import React from 'react'
import {HomeContainer} from '../styles/Home.styled.js';
import {ThreeDCubes} from './ThreeDCubes';
import {HelloButton} from './HelloButton';

export const Home = ({theme, cubesColor}) => {
    
    return (
        <HomeContainer data-testid="home">
            <div className="designs-container">
                <ThreeDCubes cubesColor={cubesColor}/>
                <HelloButton />
            </div>
            <main className="info-container">

            </main>
            <div className="quote">

            </div>
            <div className="github-icon">

            </div>
        </HomeContainer>
    )
}