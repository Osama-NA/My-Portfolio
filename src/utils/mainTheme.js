export const mainTheme = {
    themeColors: {
        aquaGreen: '#ffaaa5'
    },
    fontColors: {
        gray: '#e0e0e0',
        white: '#fff',
        hover: '#7e7bf8'
    },
    backgroundColors: {
        darkGray: '#363062',
        darkerGray: '#29244d',
        lightGray: '#4d4c7d'
    },
    cubesColors: {
        aquaGreen: '#ff7272'
    },
    circleBackground: {
        color: '#4d4c7d'
    }
}