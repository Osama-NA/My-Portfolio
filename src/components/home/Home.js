import React from 'react'
import {HomeContainer} from './styles/Home.styled.js';
import {ThreeDCubes} from './components/ThreeDCubes';
import { HelloButton } from './components/HelloButton';
import { Info } from './components/Info';
import {Quote} from './components/Quote';
import { IconLinks } from './components/IconLinks';

export const Home = ({cubesColor, circleBackground}) => {
    return (
        <HomeContainer data-testid="home">
            <ThreeDCubes cubesColor={cubesColor} />
            <Info circleBackground={circleBackground} />
            <footer>
                <Quote />
                <IconLinks />
                <HelloButton />
            </footer>
        </HomeContainer>
    )
}