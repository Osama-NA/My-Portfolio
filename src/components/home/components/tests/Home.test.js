import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import { Home } from '../Home.js';
import {ThemeProvider} from 'styled-components';
import {BrowserRouter} from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';
import { mainTheme } from '../../../../utils/mainTheme.js';

afterEach(cleanup);

describe("Home component tests", () => {

    const HomeComponent = () => {
        return (
            <BrowserRouter>
                <ThemeProvider theme={mainTheme}>
                    <Home 
                        theme={mainTheme} 
                        cubesColor="lightpink" 
                        circleBackground="lightblue"
                    />
                </ThemeProvider>
            </BrowserRouter>
        )
    }

    it("Home component renders without crashing", () => {
        let div = document.createElement('div');
        ReactDOM.render(<HomeComponent />, div);
    })

    it("Home component is in the document", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toBeInTheDocument();
    })

    it("Home component contains HelloButton component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('hello-button'));
    })

    it("Home component contains ThreeDCubes component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('threed-cubes'));
    })

    it("Home component contains Info component", () =>{
        const { getByTestId } = render(<HomeComponent />);
        expect(getByTestId('home')).toContainElement(getByTestId('info'));
    })

    it("Home component matches snapshot", () => {
        const tree = renderer.create(<HomeComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})