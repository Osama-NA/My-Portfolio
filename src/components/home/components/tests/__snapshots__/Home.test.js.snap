// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests Home component matches snapshot 1`] = `
<div
  className="sc-bdfBQB euErrF"
  data-testid="home"
>
  <div
    className="designs-container"
  >
    <div
      className="sc-gsTEea bvNmkz"
      data-testid="threed-cubes"
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <canvas
          style={
            Object {
              "display": "block",
            }
          }
        />
      </div>
    </div>
    <div
      className="sc-dlfnuX dvPYfr"
      data-testid="hello-button"
    >
      <p
        className="hello-btn"
        onClick={[Function]}
      >
        Hello there!
      </p>
      <div
        className="message-box"
        data-testid="message-box"
        style={Object {}}
      >
        <p>
          Donâ€™t waste your time clicking a hello text,
          <br />
          <a
            data-testid="message-box-link"
            href="/Router/Projects"
            onClick={[Function]}
          >
            view my work here.
          </a>
        </p>
        <i
          className="fas fa-times close-btn"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <main
    className="sc-hKgJUU keZIZm"
  >
    <h1>
      Osama Al Hajj Ali
    </h1>
    <h3>
      I am a web developer
    </h3>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Projects
        </a>
      </li>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Contact me
        </a>
      </li>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          About me
        </a>
      </li>
    </ul>
  </main>
</div>
`;
